failure to make vcd file

```
Assembly translation passed: ld.lock x30, 0(x1)
Assembly translation passed: ld.lock x30, 8(x1)
Assembly translation passed: ld.lock x30, 16(x1)
Starting RISC-V Toolchain build process

Removing existing riscv-tests/build directory
Configuring project riscv-tests
configure: WARNING: using cross tools not prefixed with host triplet
Building project riscv-tests
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone.c:20:1: warning: return type defaults to 'int' [-Wimplicit-int]
 Proc_6 (Enum_Val_Par, Enum_Ref_Par)
 ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone.c: In function 'Proc_6':
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone.c:29:9: warning: implicit declaration of function 'Func_3' [-Wimplicit-function-declaration]
   if (! Func_3 (Enum_Val_Par))
         ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone.c: At top level:
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone.c:54:1: warning: return type defaults to 'int' [-Wimplicit-int]
 Proc_7 (Int_1_Par_Val, Int_2_Par_Val, Int_Par_Ref)
 ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone.c:74:1: warning: return type defaults to 'int' [-Wimplicit-int]
 Proc_8 (Arr_1_Par_Ref, Arr_2_Par_Ref, Int_1_Par_Val, Int_2_Par_Val)
 ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c: In function 'main':
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:120:7: warning: implicit declaration of function 'Proc_5' [-Wimplicit-function-declaration]
       Proc_5();
       ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:121:7: warning: implicit declaration of function 'Proc_4' [-Wimplicit-function-declaration]
       Proc_4();
       ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:127:21: warning: implicit declaration of function 'Func_2'; did you mean 'Func_1'? [-Wimplicit-function-declaration]
       Bool_Glob = ! Func_2 (Str_1_Loc, Str_2_Loc);
                     ^~~~~~
                     Func_1
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:133:2: warning: implicit declaration of function 'Proc_7' [-Wimplicit-function-declaration]
  Proc_7 (Int_1_Loc, Int_2_Loc, &Int_3_Loc);
  ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:138:7: warning: implicit declaration of function 'Proc_8' [-Wimplicit-function-declaration]
       Proc_8 (Arr_1_Glob, Arr_2_Glob, Int_1_Loc, Int_3_Loc);
       ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:140:7: warning: implicit declaration of function 'Proc_1'; did you mean 'Func_1'? [-Wimplicit-function-declaration]
       Proc_1 (Ptr_Glob);
       ^~~~~~
       Func_1
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:147:4: warning: implicit declaration of function 'Proc_6' [-Wimplicit-function-declaration]
    Proc_6 (Ident_1, &Enum_Loc);
    ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:158:7: warning: implicit declaration of function 'Proc_2' [-Wimplicit-function-declaration]
       Proc_2 (&Int_1_Loc);
       ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c: At top level:
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:242:1: warning: return type defaults to 'int' -Wimplicit-int]
 Proc_1 (Ptr_Val_Par)
 ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c: In function 'Proc_1':
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:258:3: warning: implicit declaration of function 'Proc_3'; did you mean 'Proc_1'? [-Wimplicit-function-declaration]
   Proc_3 (&Next_Record->Ptr_Comp);
   ^~~~~~
   Proc_1
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c: At top level:
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:276:1: warning: return type defaults to 'int' -Wimplicit-int]
 Proc_2 (Int_Par_Ref)
 ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:299:1: warning: return type defaults to 'int' -Wimplicit-int]
 Proc_3 (Ptr_Ref_Par)
 ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:314:1: warning: return type defaults to 'int' -Wimplicit-int]
 Proc_4 () /* without parameters */
 ^~~~~~
/home/sercan/secchisel_v2/src/rocket-chip/riscv-tools/riscv-tests/build/../benchmarks/dhrystone/dhrystone_main.c:326:1: warning: return type defaults to 'int' -Wimplicit-int]
 Proc_5 () /* without parameters */
 ^~~~~~
Installing project riscv-tests

RISC-V Toolchain installation completed!
./emulator-freechips.rocketchip.system-DefaultConfig-debug +max-cycles=100000000 +verbose -voutput/rv64ui-p-ld_sd_lock_unlock_func.vcd output/rv64ui-p-ld_sd_lock_unlock_func 3>&1 1>&2 2>&3 | /home/sercan/secchisel_v2/src/rocket-chip/riscv/bin/spike-dasm  > output/rv64ui-p-ld_sd_lock_unlock_func.out && [ $PIPESTATUS -eq 0 ]
Makefile:43: recipe for target 'output/rv64ui-p-ld_sd_lock_unlock_func.vcd' failed
make: *** [output/rv64ui-p-ld_sd_lock_unlock_func.vcd] Error 1
```

We have been trying to test our instructions. One of them is ld.lock. 

Test is executed without error while only testing this case.
`TEST_LD_SEC_OP( 2, 0x00ff00ff00ff00ff, tdat, ld.lock x30, 0(x1) );`

Test is also executed without error while only testing this two cases.
```
TEST_LD_SEC_OP( 2, 0x00ff00ff00ff00ff, tdat, ld.lock x30, 0(x1) );
TEST_LD_SEC_OP( 3, 0xff00ff00ff00ff00, tdat, ld.lock x30, 8(x1) );
```

Also, test is executed without error while only testing this case.
`TEST_LD_SEC_OP( 4, 0x0ff00ff00ff00ff0, tdat, ld.lock x30, 16(x1) );`

However, when we tried to test with these three cases together, we had error below.
```

TEST_LD_SEC_OP( 2, 0x00ff00ff00ff00ff, tdat, ld.lock x30, 0(x1) );
TEST_LD_SEC_OP( 3, 0xff00ff00ff00ff00, tdat, ld.lock x30, 8(x1) );
TEST_LD_SEC_OP( 4, 0x0ff00ff00ff00ff0, tdat, ld.lock x30, 16(x1) );
```

```
Makefile:43: recipe for target 'output/rv64ui-p-ld_sd_lock_unlock_func.vcd' failed
make: *** [output/rv64ui-p-ld_sd_lock_unlock_func.vcd] Error 1
```

--- a/csrc/emulator.cc
+++ b/csrc/emulator.cc
@@ -313,17 +313,17 @@ done_processing:
 
   if (dtm->exit_code())
   {
-    fprintf(stderr, "*** FAILED *** (code = %d, seed %d) after %ld cycles\n", dtm->exit_code(), random_seed, trace_count);
+    fprintf(stderr, "*** FAILED *** via dtm (code = %d, seed %d) after %ld cycles\n", dtm->exit_code(), random_seed, trace_count);
     ret = dtm->exit_code();
   }
   else if (jtag->exit_code())
   {
-    fprintf(stderr, "*** FAILED *** (code = %d, seed %d) after %ld cycles\n", jtag->exit_code(), random_seed, trace_count);
+    fprintf(stderr, "*** FAILED *** via jtag (code = %d, seed %d) after %ld cycles\n", jtag->exit_code(), random_seed, trace_count);
     ret = jtag->exit_code();
   }
   else if (trace_count == max_cycles)
   {
-    fprintf(stderr, "*** FAILED *** (timeout, seed %d) after %ld cycles\n", random_seed, trace_count);
+    fprintf(stderr, "*** FAILED *** via trace_count (timeout, seed %d) after %ld cycles\n", random_seed, trace_count);
     ret = 2;
   }
   else if (verbose || print_cycles)

--- a/csrc/remote_bitbang.cc
+++ b/csrc/remote_bitbang.cc
@@ -20,7 +20,8 @@ remote_bitbang_t::remote_bitbang_t(uint16_t port) :
   socket_fd(0),
   client_fd(0),
   recv_start(0),
-  recv_end(0)
+  recv_end(0),
+  err(0)
 {
   socket_fd = socket(AF_INET, SOCK_STREAM, 0);
   if (socket_fd == -1) {
@@ -61,7 +62,7 @@ remote_bitbang_t::remote_bitbang_t(uint16_t port) :
     fprintf(stderr, "remote_bitbang getsockname failed: %s (%d)\n",
             strerror(errno), errno);
     abort();
-  } 
+  }
 
   tck = 1;
   tms = 1;

